src/wasm-interpreter.h

```diff
@@ -2228,6 +2228,186 @@ class ModuleInstance
     : ModuleInstanceBase(wasm, externalInterface) {}
 };
 
+static const Name NONSTANDALONE_FLOW("Binaryen|nonstandalone");
+
+typedef std::unordered_map<LocalGet*, Literals> GetValues;
+
+// Evaluates a standalone expression. Errors if we hit anything that can't be
+// evaluated.
+class StandaloneExpressionRunner
+  : public ExpressionRunner<StandaloneExpressionRunner> {
+  Module* module;
+
+  // map gets to constant values, if they are known to be constant
+  GetValues& getValues;
+  // map local indexes to values set in the context of this runner
+  std::unordered_map<Index, Literals> setLocalValues;
+  // map global names to values set in the context of this runner
+  std::unordered_map<Name, Literals> setGlobalValues;
```

Comment:
IIUC, `getValues` is a mapping of specific gets to values, while `setLocalValues` maps indexes to values, so it applies to all gets for that index? That seems somewhat redundant.

One option might be to have just the former, and e.g. `visit()` could receive optional maps of `index => value` that it would then use to scan for gets and mark them with the proper value using the single mechanism.

Alternatively maybe we can rename/document this better so it's clearer.

Reply:
From what I understood `getValues` is special in that it is referenced from the precompute pass and is possibly very large, since it covers all the gets up to the point where the runner was created. So, copying from there to a new map or vice-versa might involve quite some overhead. Going to see what can be done.

Reply:
So it seems that no matter how we turn it, if we'd like to unify both concepts we'll end up polluting `getValues` of the precompute pass, except if we copy the whole map beforehand, which appears to be intentionally avoided because it can be large. What I did now is make `getValues` optional, as a pointer (which was necessary anyway for the C API), still taking precedence over manually set values as it's a pass-only mechanism (either or) anyway. Does that sound reasonable?